src/materials/nodes/SpriteNodeMaterial.js

```diff
@@ -47,14 +50,27 @@ class SpriteNodeMaterial extends NodeMaterial {
 
 		let mvPosition = modelViewMatrix.mul( vec3( positionNode || 0 ) );
 
-		let scale = vec2( modelWorldMatrix[ 0 ].xyz.length(), modelWorldMatrix[ 1 ].xyz.length() );
+		let scale = vec2( modelWorldMatrix[ 0 ].xyz.length(), modelWorldMatrix[ 1 ].xyz.length() ).toVar();
 
 		if ( scaleNode !== null ) {
 
 			scale = scale.mul( scaleNode );
 
 		}
 
+
+		if ( ! useSizeAttenuation ) {
+
+			const perspective = cameraProjectionMatrix.element( 2 ).element( 3 ).equal( - 1.0 );
```

Comment:
Can we use an existing method to identify a perspective camera?

@sunag may have a preference.

Reply:
We can get the current camera in the `NodeBuilder`, `setupPosition( { ..., camera } )`, it would be better to do it in JS.